main.py

```diff
@@ -11,11 +11,12 @@
 import pathlib
 from functools import partial
 
-from src import run
+from src import run, types_
 from src.discourse import create_discourse
 
 
-def main():
+# pylint: disable=too-many-locals
```

Comment:
I do agree that it would be beneficial to break these functions down into
1. Input arguments parsing
2. Execution
3. Output
This would greatly help with upcoming issues such as  supporting cli as well!
Thank you for the comment